musig/README.md

```diff
@@ -464,3 +486,14 @@ Operation:
 
 Output:
 - `Result<Scalar, VMError>`
+
+
+## Modifications from the paper
+
+### Using Merlin transcripts for hashing and randomness
+
+The paper uses multiple hashing algorithms (`H_agg` and `H_sig` in the multikey Musig protocol, and `H` in the multimessage protocol). We instead use Merlin transcripts, by feeding the expected hash input to the transcript, and getting a challenge from the transcript instead of a hash output. The benefits of using transcripts is detailed in this [blog post about Merlin transcripts](https://medium.com/@hdevalence/merlin-flexible-composable-transcripts-for-zero-knowledge-proofs-28d9fda22d9a).
+
+### Changing challenge definition for Multikey case
+
+In order to have a consistent multi-party signer protocol between the Multikey and Multimessage cases, we changed the definition of the hash function `H_agg` in the Multikey case from `c_i = H_agg(<L>, X_i)` to `c_i = H_agg(<L>, i)`. This way, the `.challenge(...)` functions in the Multikey and Multimessage cases are dependent on the public key's index `i`, instead of the public key itself. The safety of the protocol is not changed, since the hash function `H_agg` also takes `<L>`, the encoding of the multiset of the the public keys, as an input, so the hash output is bound to a specific public key in the list both when it takes in `i` and `X_i`.
```

Comment:
```suggestion
In order to have a consistent multi-party signer protocol between the Multikey and Multimessage cases, we changed the definition of the hash function `H_agg` in the Multikey case from `c_i = H_agg(<L>, X_i)` to `c_i = H_agg(<L>, i)`. This way, the `.challenge(...)` functions in the Multikey and Multimessage cases are dependent on the public key's index `i`, instead of the public key itself. The safety of the protocol is not changed, since the hash function `H_agg` also takes `<L>`, the encoding of the multiset of the the public keys, as an input, so the hash output is bound to a specific public key in the list both when it takes in `i` and `X_i`.

```